src/libsyntax/attr.rs

```diff
@@ -877,10 +886,38 @@ pub fn find_repr_attrs(diagnostic: &Handler, attr: &Attribute) -> Vec<ReprAttr>
                     if let Some(h) = hint {
                         acc.push(h);
                     }
+                } else if let Some((name, value)) = item.name_value_str() {
+                    if name == "align" {
+                        use core::str::FromStr;
+                        if let Ok(align) = u64::from_str(&*value.as_str()) {
+                            if align.is_power_of_two() {
+                                // rustc::ty::layout::Align restricts align to <= 32768
+                                if align <= 32768 {
+                                    acc.push(ReprAlign(align));
+                                } else {
+                                    span_err!(diagnostic, item.span, E0586,
+                                              "alignment representation hint is greater than 2^15");
+                                }
+                            } else {
+                                span_err!(diagnostic, item.span, E0585,
+                                          "alignment representation hint is not a power of two");
+                            }
+                        } else {
+                            span_err!(diagnostic, item.span, E0584,
+                                      "alignment representation hint is not a u64");
```

Comment:
Oh, we could probably use a single error message for an invalid `#[repr(align)]` (also, "hint" is probably the wrong thing to call it?).